lib/util/flattenOperands.js

```diff
@@ -86,7 +86,9 @@ function flattenOperands(node) {
     return node;
   }
   else if (Node.Type.isUnaryMinus(node)) {
-    const arg = flattenOperands(node.args[0]);
+    // since arg is changed when it is negated, clone it before negation to avoid
+    // modifiying original reference
+    const arg = flattenOperands(node.args[0].cloneDeep());
```

Comment:
Here we find the real pain. Note how the arg gets negated in the next line. When this is not cloned, it ends up messing up with the original node in stepThrough.

if you boot up your node console and try solving `-((1)/(3))=((-1)/(3))` you'll see that on the left side an _extra_ negative gets added! gasp! this was such a pain to track down. 

For background, this was occuring in `simplifyExpression/stepThrough` in the `step` function 

see L47:
```
console.log(node.toString()) // my logger
nodeStatus = step(node);
console.log(node.toString()) // my logger
```

gasp!